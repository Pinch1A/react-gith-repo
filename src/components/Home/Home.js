import { Button } from '../../App'
import React from 'react'
import styled from 'styled-components'

const HomePage = styled.main`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 4rem;
`

const Header = styled.header`
  margin-bottom: 4rem;
  text-align: center;
  font-size: 1.5rem;
`

const Title = styled.h3`
  font-size: 3rem;
  font-weight: bold;
  margin-bottom: 2rem;
`

const ListWrapper = styled.div`
  padding: 1rem;
  position: relative;
`
const List = styled.ul`
  padding: 1rem;
`
const ListElem = styled.li`
  margin-bottom: 2rem;
`

export const Home = ({ history }) => {
  const nextPage = () => {
    history.push('/token')
  }
  return (
    <HomePage>
      <Header>
        <Title>GitHub Repo Finder</Title>
        <p>
          A simple application which interact with GitHub API V4 with the help
          of Access token
        </p>
      </Header>

      <ListWrapper>
        <List>
          <ListElem>Validate your Access token</ListElem>
          <ListElem>Find a GitHub Public Repo</ListElem>
          <ListElem>Provides the repo details of Results</ListElem>
        </List>
      </ListWrapper>
      <Button primary onClick={nextPage}>
        Go!{' '}
      </Button>
    </HomePage>
  )
}
