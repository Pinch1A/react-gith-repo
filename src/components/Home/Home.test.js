import { Home } from './Home'
import React from 'react'
import { shallow } from 'enzyme'

const props = {
  history: jest.fn(),
}

describe('<Home>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Home {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
