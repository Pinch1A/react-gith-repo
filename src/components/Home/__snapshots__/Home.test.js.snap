// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> renders without crashing 1`] = `
<styled.main>
  <styled.header>
    <styled.h3>
      GitHub Repo Finder
    </styled.h3>
    <p>
      A simple application which interact with GitHub API V4 with the help of Access token
    </p>
  </styled.header>
  <styled.div>
    <styled.ul>
      <styled.li>
        Validate your Access token
      </styled.li>
      <styled.li>
        Find a GitHub Public Repo
      </styled.li>
      <styled.li>
        Provides the repo details of Results
      </styled.li>
    </styled.ul>
  </styled.div>
  <styled.button
    onClick={[Function]}
    primary={true}
  >
    Go!
     
  </styled.button>
</styled.main>
`;
